specs/Standards/Sould Bound Token/README.md

```diff
@@ -0,0 +1,458 @@
+    pub reference: Option<String>, // URL to a JSON file with more info
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+/// TokenMetadata defines attributes for each SBT token.
+pub struct TokenMetadata {
+    pub class: ClassId, // token class
+    pub issued_at: Option<u64>, // When token was issued or minted, Unix epoch in milliseconds
+    pub expires_at: Option<u64>, // When token expires, Unix epoch in milliseconds
+    pub reference: Option<String>, // URL to an off-chain JSON file with more info.
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+
+trait SBTRegistry {
+    /**********
+    * QUERIES
+    **********/
+
+    /// get the information about specific token ID issued by `ctr` SBT contract.
+    fn sbt(&self, ctr: AccountId, token: TokenId) -> Option<Token>;
+
+    /// returns total amount of tokens issued by `ctr` SBT contract.
+    fn sbt_supply(&self, ctr: AccountId) -> u64;
+
+    /// returns total amount of tokens of given class minted by this contract
+    fn sbt_supply_by_class(&self, ctr: AccountId, class: ClassId) -> u64;
+
+    /// returns total supply of SBTs for a given owner.
+    /// If class is specified, returns only owner supply of the given class -- must be 0 or 1.
+    fn sbt_supply_by_owner(
+        &self,
+        ctr: AccountId,
+        account: AccountId,
+        class: Option<ClassId>,
+    ) -> u64;
+
+    /// Query sbt tokens issued by a given contract.
+    /// If `from_index` is not specified, then `from_index` should be assumed
+    /// to be the first valid token id.
+    fn sbt_tokens(
+        &self,
+        ctr: AccountId,
+        from_index: Option<u64>,
+        limit: Option<u32>,
+    ) -> Vec<TokenId>;
+
+    /// Query SBT tokens by owner
+    /// If `from_class` is not specified, then `from_class` should be assumed to be the first
+    /// valid class id.
+    /// Returns list of pairs: `(Contract address, list of token IDs)`.
+    fn sbt_tokens_by_owner(
+        &self,
+        account: AccountId,
+        ctr: Option<AccountId>,
+        from_class: Option<u64>,
+        limit: Option<u32>,
+    ) -> Vec<(AccountId, Vec<TokenId>)>;
+
+    /// checks if an `account` was banned by the registry.
+    fn is_banned(&self, account: AccountId) -> bool;
+
+
+    /*************
+     * Transactions
+     *************/
+
+    /// Creates a new, unique token and assigns it to the `receiver`.
+    /// `token_spec` is a vector of pairs: owner AccountId and TokenMetadata.
+    /// Each TokenMetadata must have non zero `class`.
+    /// Must be called by an SBT contract.
+    /// Must emit `Mint` event.
+    /// Must provide enough NEAR to cover registry storage cost.
+    // #[payable]
+    fn sbt_mint(&mut self, token_spec: Vec<(AccountId, TokenMetadata)>) -> Vec<TokenId>;
```

Comment:
Is there a meaningful mint implementation that anyone can call? NFT and FT standards do not define `mint` interface because usually there is no meaningful way for a random user out there to mint a new token on their own, and then there is no reason to define mint method as part of the required specification as mints could be batched or executed only as a callback.

Reply:
This is the registry mint, not the SBT mint. Since every SBT smart contract, must agree how to use the registry, we need it in the standard. However, it SBT smart contract mint is not part of the standard, for the same reasons you mentioned.